qiskit/qasm/qasmlexer.py

```diff
@@ -160,7 +160,7 @@ def t_INCLUDE(self, _):
         return self.lexer.token()
 
     def t_FORMAT(self, t):
-        r"OPENQASM\s+(\d+)\.(\d+)"
+        r"OPENQASM\s+2\.0"
         return t
```

Comment:
I think it would be good if we update the error messages such that if you try and parse a programme beginning with `OPENQASM 3.0;` or similar, the error tells you that this module is for QASM 2 only, and add a test that that works correctly.  This might require keeping the token the way it currently is, and handling it later on, during parsing.  At any rate, it'd be good to clean up `QasmParser.p_format_0`, either by absorbing it into `p_format` to actually perform the check, or removing it if we're not going to use it at all.